nifi-docs/src/main/asciidoc/walkthroughs.adoc

```diff
@@ -0,0 +1,1099 @@
+Serial number: 0
+Valid from: Mon Feb 08 20:24:06 PST 2016 until: Thu Feb 05 20:24:06 PST 2026
+Certificate fingerprints:
+	 SHA1: E9:CA:EF:26:C7:2A:AF:7A:FD:BC:97:12:ED:F5:6B:3B:79:40:D2:A3
+	 SHA256: 4C:0A:2F:C4:D3:CC:B5:A8:0E:0B:95:EE:0C:E6:60:7C:85:78:7A:58:77:BB:87:58:12:9B:D1:E4:BA:E0:9B:67
+Signature algorithm name: SHA256withRSA
+Subject Public Key Algorithm: 2048-bit RSA key
+Version: 3
+
+Extensions:
+
+#1: ObjectId: 2.5.29.19 Criticality=true
+BasicConstraints:[
+  CA:true
+  PathLen:2147483647
+]
+
+#2: ObjectId: 2.5.29.31 Criticality=false
+CRLDistributionPoints [
+  [DistributionPoint:
+     [URIName: http://crl.tinycert.org/ca-2405.crl]
+]]
+
+#3: ObjectId: 2.5.29.18 Criticality=false
+IssuerAlternativeName [
+  RFC822Name: andy@andylopresto.com
+]
+
+#4: ObjectId: 2.5.29.15 Criticality=true
+KeyUsage [
+  DigitalSignature
+  Key_Encipherment
+  Key_Agreement
+  Key_CertSign
+]
+
+#5: ObjectId: 2.5.29.17 Criticality=false
+SubjectAlternativeName [
+  RFC822Name: andy@andylopresto.com
+]
+
+#6: ObjectId: 2.5.29.14 Criticality=false
+SubjectKeyIdentifier [
+KeyIdentifier [
+0000: 64 A2 F6 E9 F1 1A 9B 0F   40 5E 0F 23 DC AC 16 20  d.......@^.#...
+0010: 6B 5A F3 74                                        kZ.t
+]
+]
+
+Trust this certificate? [no]:  yes
+Certificate was added to keystore
+----
+--
+. Optionally move the `keystore.jks` and `truststore.jks` files into the `conf/` directory. This step is not required, but helps with organization.
+. Configure the `nifi.properties` file with the necessary values. This includes setting the hostname, ports, and keystore and truststore paths and passwords.
+* `$EDITOR conf/nifi.properties` -- Opens the `nifi.properties` file in a text editor
+* Update the following lines:
+** `nifi.remote.input.host=` -> `nifi.remote.input.host=secure.nifi`
+** `nifi.remote.input.secure=false` -> `nifi.remote.input.secure=true`
+** `nifi.remote.input.socket.port=` -> `nifi.remote.input.socket.port=10443`
+** `nifi.web.http.port=8080` -> `nifi.web.http.port=`
+** `nifi.web.https.host=` -> `nifi.web.https.host=secure.nifi`
+** `nifi.web.https.port=` -> `nifi.web.https.port=9443`
+** `nifi.security.keystore=` -> `nifi.security.keystore=./conf/keystore.jks`
+** `nifi.security.keystoreType=` -> `nifi.security.keystoreType=jks`
+** `nifi.security.keystorePasswd=` -> `nifi.security.keystorePasswd=passwordpassword`
+** `nifi.security.truststore=` -> `nifi.security.truststore=./conf/truststore.jks`
+** `nifi.security.truststoreType=` -> `nifi.security.truststoreType=jks`
+** `nifi.security.truststorePasswd=` -> `nifi.security.truststorePasswd=passwordpassword`
+. Generate the client certificate keystore from the client certificate and key.
+`openssl pkcs12 -export -out CN=my_username.p12 -inkey client.key -in client.pem` -- Creates the PKCS12 keystore containing the client certificate and private key. This command prompts for an *export password* twice
```

Comment:
For consistency with previous examples, the open.ssl command should be a separate bullet point.